ext/date/php_date.c

```diff
@@ -311,16 +311,18 @@ static void date_throw_uninitialized_error(zend_class_entry *ce)
 {
 	if (ce->type == ZEND_INTERNAL_CLASS) {
 		zend_throw_error(date_ce_date_object_error, "Object of type %s has not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name));
-	} else {
-		zend_class_entry *ce_ptr = ce;
-		while (ce_ptr && ce_ptr->parent && ce_ptr->type == ZEND_USER_CLASS) {
-			ce_ptr = ce_ptr->parent;
-		}
-		if (ce_ptr->type != ZEND_INTERNAL_CLASS) {
-			zend_throw_error(date_ce_date_object_error, "Object of type %s not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name));
-		}
-		zend_throw_error(date_ce_date_object_error, "Object of type %s (inheriting %s) has not been correctly initialized by calling parent::__construct() in its constructor", ZSTR_VAL(ce->name), ZSTR_VAL(ce_ptr->name));
+		return;
+	}
+
+	zend_class_entry *ce_ptr = ce;
```

Comment:
Yes: https://github.com/php/php-src/pull/15598#discussion_r1734052157 Although I commented this to the wrong line, because line 320 (before my changes) was missing a `return` and therefore `zend_throw_error` was invoked twice

Reply:
It is unrelated to the actual change though (memory reduction); I also don't think the code style *should* be updated in the same PR/commit, and even if you did, I would prefer that variable declarations be kept at the top of functions.